s3torchconnectorclient/rust/src/logger_setup.rs

```diff
@@ -3,62 +3,54 @@
  * // SPDX-License-Identifier: BSD
  */
 use std::{env};
-use log::{LevelFilter};
 use mountpoint_s3_crt::common::rust_log_adapter::RustLogAdapter;
 use pyo3::{PyResult};
-use pyo3_log::Logger;
 use tracing_subscriber::{filter::EnvFilter};
 use tracing_subscriber::util::{SubscriberInitExt};
 use crate::exception::python_exception;
 
-pub const ENABLE_CRT_LOGS_ENV_VAR: &str = "ENABLE_CRT_LOGS";
-pub const CRT_LOGS_DIR_PATH_ENV_VAR: &str = "CRT_LOGS_DIR_PATH";
+pub const ENABLE_CRT_LOGS_ENV_VAR: &str = "S3_CONNECTOR_ENABLE_CRT_LOGS";
+pub const LOGS_DIR_PATH_ENV_VAR: &str = "S3_CONNECTOR_LOGS_DIR_PATH";
 
 pub fn setup_logging() -> PyResult<()> {
     let enable_crt_logs = env::var(ENABLE_CRT_LOGS_ENV_VAR);
+    let mut filter= EnvFilter::from_default_env();
 
     if enable_crt_logs.is_ok() {
-        enable_crt_logging()
-    } else {
-        enable_default_logging()
+        RustLogAdapter::try_init().map_err(python_exception)?;
+        filter = EnvFilter::from_env(ENABLE_CRT_LOGS_ENV_VAR);
     }
-}
-
-fn enable_crt_logging() -> PyResult<()> {
-    RustLogAdapter::try_init().map_err(python_exception)?;
 
-    let filter = EnvFilter::try_from_env(ENABLE_CRT_LOGS_ENV_VAR).map_err(python_exception)?;
-    let crt_logs_path = env::var(CRT_LOGS_DIR_PATH_ENV_VAR).ok();
+    let crt_logs_path = env::var(LOGS_DIR_PATH_ENV_VAR).ok();
 
     match crt_logs_path {
         Some(logs_path) => {
-            let logfile = tracing_appender::rolling::hourly(logs_path, "s3torchconnectorclient.log");
-            let subscriber_builder = tracing_subscriber::fmt()
-                .with_writer(logfile)
-                .with_env_filter(filter)
-                .with_ansi(false);
-            subscriber_builder.finish().try_init().map_err(python_exception)?;
-        },
+            enable_file_logging(filter, logs_path)?;
+        }
         None => {
-            let subscriber_builder = tracing_subscriber::fmt()
-                 .with_env_filter(filter)
-                 .with_ansi(false);
-            subscriber_builder.finish().try_init().map_err(python_exception)?;
+            enable_default_logging(filter)?;
         }
     }
 
     Ok(())
 }
 
-fn enable_default_logging() -> PyResult<()> {
-    let logger = Logger::default()
-        .filter_target(
-            "mountpoint_s3_client::s3_crt_client::request".to_owned(),
-            LevelFilter::Off,
-        )
-        .filter(LevelFilter::Trace);
+fn enable_file_logging(filter: EnvFilter, logs_path: String) -> PyResult<()> {
+    let logfile = tracing_appender::rolling::hourly(logs_path, "s3torchconnectorclient.log");
```

Comment:
Why did we call it `s3torchconnectorclient`?

Reply:
I wanted to avoid s3torchconnector.log as that's handled by the Python logger and I did not want to confuse things more.